specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json

```diff
@@ -0,0 +1,641 @@
+            "$ref": "#/definitions/FormOperationError"
+          }
+        }
+      }
+    },
+    "KeysResult": {
+      "description": "Result of an operation to get\r\nthe keys extracted by a model.",
+      "type": "object",
+      "properties": {
+        "clusters": {
+          "description": "Object mapping ClusterIds to Key lists.",
+          "type": "object",
+          "additionalProperties": {
+            "uniqueItems": false,
+            "type": "array",
+            "items": { "type": "string" }
+          }
+        }
+      }
+    },
+    "ModelResult": {
+      "description": "Result of a model status query operation.",
+      "type": "object",
+      "properties": {
+        "modelId": {
+          "description": "Get or set model identifier.",
+          "type": "string",
+          "format": "uuid",
+          "x-nullable": false
+        },
+        "status": {
+          "description": "Get or set the status of model.",
+          "enum": [ "created", "ready", "invalid" ],
+          "type": "string"
+        },
+        "createdDateTime": {
+          "format": "date-time",
+          "description": "Get or set the created date time of the model.",
+          "type": "string"
+        },
+        "lastUpdatedDateTime": {
+          "format": "date-time",
+          "description": "Get or set the model last updated datetime.",
+          "type": "string"
+        }
+      }
+    },
+    "ModelsResult": {
+      "description": "Result of query operation to fetch multiple models.",
+      "type": "object",
+      "properties": {
+        "models": {
+          "description": "Collection of models.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ModelResult"
+          }
+        }
+      }
+    },
+    "FormDocumentReport": {
+      "type": "object",
+      "properties": {
+        "documentName": {
+          "description": "Reference to the data that the report is for.",
+          "type": "string"
+        },
+        "pages": {
+          "format": "int32",
+          "description": "Total number of pages trained on.",
+          "type": "integer"
+        },
+        "errors": {
+          "description": "List of errors per page.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "status": {
+          "description": "Status of the training operation.",
+          "enum": [
+            "success",
+            "partialSuccess",
+            "failure"
+          ],
+          "type": "string"
+        }
+      }
+    },
+    "FormOperationError": {
+      "description": "Error reported during an operation.",
+      "type": "object",
+      "properties": {
+        "errorMessage": {
+          "description": "Message reported during the train operation.",
+          "type": "string"
+        }
+      }
+    },
+    "ErrorResponse": {
```

Comment:
@ramparab have you considered modeling your error response to something like 
```
{
    "error": {
        "code": "",
        "message": "",
        "target": "",
        "additionalInfo": [],
        "details": [
            {
                "code": "",
                "target": "",
               "message": "",
                "additionalInfo": [
                    {
                        "type": "PolicyViolation",
                        "info": {
                            "policySetDefinitionDisplayName": "Secure the environment",
                            "policySetDefinitionId":"/subscriptions/7400bec1d56043e88080ab09fc9c02cb/providers/Microsoft.Authorization/policySetDefinitions/TestPolicySet",
                            "policyDefinitionDisplayName": "Allowed locations",
                            "policyDefinitionId":"/subscriptions/7400bec1d56043e88080ab09fc9c02cb/providers/Microsoft.Authorization/policyDefinitions/TestPolicy1",
                            "policyAssignmentDisplayName": "Allow Central US and WEU only",
                            "policyAsssignmentId":"/subscriptions/7400bec1d56043e88080ab09fc9c02cb/providers/Microsoft.Authorization/policyAssignments/TestAssignment1",
                            ...
                            ...
                            ...
                        }
                    }
                ]
            }
        ]
    }
}
```
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content
The format I am recommending allows you to have multiple error information objects as an array and will help

Reply:
Thats a fair point. We were on very short schedule so we did not have much time to invest in this. We do intend to address this in our future release. Please let me know if this is a blocker.